bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingLinkManager.java

```diff
@@ -163,8 +163,10 @@ public void removed(ItemChannelLink itemChannelLink) {
 
         @Override
         public void updated(ItemChannelLink oldElement, ItemChannelLink element) {
-            this.removed(oldElement);
-            this.added(element);
+            if(oldElement != null && !oldElement.equals(element)) {
```

Comment:
Neither "oldElement", nor "element" should be null here. If that is the case, we have to fix it in the Registries, too. Otherwise other handlers will have the same problem again.